package.json

```diff
@@ -58,7 +61,7 @@
   "bugs": {
     "url": "https://github.com/softonic/axios-retry/issues"
   },
-  "types": "lib/esm/index.d.ts",
+  "types": "index.d.ts",
```

Comment:
This project is CommonJS, so index.d.ts should have been the default, not the type definition under esm, so I fixed it.

Reply:
1st of all, thank you for your commitment @yutak23 and @rchl for your feedback

This lib is actually written using ESM (`es/index.mjs`) and then built with babel for cjs and esm

I believe we followed [this guide](https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html ) back in the day to implement such compatibility.

Looking at that, I still think is a good approach and may be worth follow the TS part and migrate the implementation to TS.

So my proposal would be to: 
- revert #250 and publish as 3.9.1 to be back to the previous situation (reopen #159) as @rchl is suggesting
- work on the migration to TS: I'm not sure when I will be able to do it, I'll try this weekend

What do you guys think? Should that solve both issues #159 and the new one?

Reply:
I agree that reverting is the best way forward for now.

Those changes just raise too many questions.

Reply:
> Looking at that, I still think is a good approach and may be worth follow the TS part and migrate the implementation to TS.

Migrate to TS seems easiest in the end. I agree that moving to TS would solve the problem.
And it was not a good idea to make major changes without testing against TypeScript type definitions. I apologies.

@rchl 
Thank you for the pointers, etc.